### PR TITLE
Fix migration error causing to lose the existing redux state

### LINKED ISSUE #148: User was asked to restore account, when account was already set-up before
### Expected Behavior

This should never happen

### Current Behavior

User was asked to restore account, when account was already set-up before

It's important, because if user has not backed up phrase before, the funds could be lost.

Talk to @asaj for more details.

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This PR addresses the issue described in #148 which was caused by an error during the redux state migration.

This was initially hard to see because `redux-persist` [disables logs in production](https://github.com/rt2zz/redux-persist/blob/d7efde9115a0bd2d6a0309ac6fb1c018bf06dc30/src/persistReducer.js#L136-L137) and the error was swallowed silently 😅 

To prevent this type of error to come back in the future, the following things were done:
- Added a unit test which makes sure all migrations work from the very first state shape we had. Though this is not entirely fool proof, it already caught a similar error which we had.
- Logs are now captured during migrations.
- Errors during migrations are captured in Sentry.

### Other changes

Fix console logs not fully displayed when running jest (only the first passed argument was displayed).

### Tested

Added new tests.

### Related issues

- Fixes #148

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/redux/migrations.ts [hunk 1]
```diff
@@ -117,7 +117,7 @@ export const migrations = {
     }
   },
   8: (state: any) => {
-    const lastUsedProvider = state.fiatExchanges.lastUsedProvider
+    const lastUsedProvider = state.fiatExchanges?.lastUsedProvider
```
Comment (jeanregisser) [inline packages/mobile/src/redux/migrations.ts:120]: This is the first migration error I caught locally.

File: packages/mobile/src/redux/migrations.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export const migrations = {
     }
   },
   7: (state: any) => {
-    const newAddressToDisplayName = Object.keys(state.identity.addressToDisplayName).reduce(
+    const newAddressToDisplayName = Object.keys(state.identity.addressToDisplayName || {}).reduce(
```
Comment (jeanregisser) [inline packages/mobile/src/redux/migrations.ts:101]: This is the 2nd error we had, which was caught by the unit test.

File: packages/mobile/src/redux/createMigrate.test.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    const migratedSchema = await migrate(undefined, 0)
+    expect(testMigrations[0]).not.toHaveBeenCalled()
+    expect(testMigrations[1]).not.toHaveBeenCalled()
+    expect(testMigrations[2]).not.toHaveBeenCalled()
+    expect(testMigrations[3]).not.toHaveBeenCalled()
+    expect(migratedSchema).toBe(undefined)
+  })
+
+  // This is what the original createMigrate from redux-persist does,
+  // but should we throw instead?
```
Comment (gnardini) [inline packages/mobile/src/redux/createMigrate.test.ts:41]: I don't think it's a common enough case for it to matter, is there even a way to downgrade using the app/play store?

**Review by gnardini — APPROVED** [review]
Amazing work on this! 🙌

File: packages/mobile/src/redux/createMigrate.test.ts (hunk 3 above)
Comment (jeanregisser) [reply packages/mobile/src/redux/createMigrate.test.ts:41]: agreed, I don't know of any way for App Store/Play Store users to downgrade.

**Pull request merged**